From 435e5b93b6b1fd9d9187a1c5af9caeaeef678b6c Mon Sep 17 00:00:00 2001
From: Billy Tsai <billy_tsai@aspeedtech.com>
Date: Wed, 20 Oct 2021 18:36:03 +0800
Subject: [PATCH 1/3] peci: aspeed: Clear the error event flag.

The error event flag needs to be cleared to avoid getting error return on
the next success transfer.

Signed-off-by: Billy Tsai <billy_tsai@aspeedtech.com>
Change-Id: If7652c6642318b4a9979497d8b78a646c595901a
---
 drivers/peci/peci_aspeed.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/peci/peci_aspeed.c b/drivers/peci/peci_aspeed.c
index 66f0a88416..d43bf83a9e 100644
--- a/drivers/peci/peci_aspeed.c
+++ b/drivers/peci/peci_aspeed.c
@@ -279,6 +279,7 @@ static int peci_aspeed_transfer(const struct device *dev, struct peci_msg *msg)
 			}
 			LOG_ERR("osError: %d\n", ret);
 		} else {
+			osEventFlagsClear(peci_data->evt_id, ret);
 			if (ret & BIT(PECI_INT_W_FCS_ABORT)) {
 				LOG_ERR("FCS Abort\n");
 				return -EOPNOTSUPP;
-- 
2.25.1

